docs/devel/consumers-claims.md

```diff
@@ -0,0 +1,297 @@
+                       +----------+    |
+                             ^         |
+                             |         +---- user-created
+                             |         |
+                       +----------+    |
+                       | consumer | <--+
+                       +----------+
+```
+
+In this model a `consumer` object (created by a user in the API) files a
+`claim` (also created by the user) for some resource.  Claims are abstract by
+nature (see the storage and networking forms of this diagram below) and
+describe what kind of resource is needed, but (generally) not the identity of
+the specific resource instance.  A `Manager` entity (a process, not part of the
+API) is responsible for satisfying the claim by "binding" the claim to a
+specific resource.
+
+You can sort of see this same pattern in the scheduling of `Pod`s onto `Node`s.
+A `Pod` is a claim against resources on a `Node`, and is bound by the scheduler.
+
+## Storage
+
+Storage in Kubernetes will be represented by `PersistentVolume`s.  A user can
+create a `PersistentVolumeClaim` for storage and use that claim from a `Pod`.
+A `VolumeManager` is responsible for binding `PersistentVolumeClaim`s to
+`PersistentVolume`s.  If the `PersistentVolumeClaim` can not be satisified by
+any existing `PersistentVolume`, the `VolumeManager` may provision a new one.
+
+```
+                       +------------------+
+        +--create----> | PersistentVolume |
+        |              +------------------+
+   +----+----+               ^
+   | VolMgr  |---bind------> :
+   +----+----+               v
+        |              +------------------------+
+        +---watch----> | PersistentVolumeClaim  | <--+
+                       +------------------------+    |
+                                 ^                   |
+                                 |         +---------+-- user-created
+                                 |         |
+                           +----------+    |
+                           |    Pod   | <--+
+                           +----------+
+```
+
+### PersistentVolumeClaim
+
+A `PersistentVolumeClaim` does not specify which specific PersistentVolume it
+wants or even which volume plugin should be used.  Instead it specifies
+characteristics of the storage it needs and those characteristics are satisfied
+as best is possible.  Characteristics include the minimum storage space
+required and the minimum access mode required (e.g. multi-writer).
+
+As the ecosystem of volume plugins grows, there will be an ever-increasing
+number of things that could be specified about volumes.  Rather than growing
+`PersistentVolumeClaim` to cover the union of every volume's parameters, users
+will be able to request coarse "classes" of volume (e.g. Gold, Silver, Bronze).
+How those classes map to real storage technologies is a policy set by the
+cluster administrators.
+
+### VolumeManagers
+
+The act of binding a `PersistentVolumeClaim` to a `PersistentVolume` is the
```

Comment:
Maybe say something about the relationship between the scheduler binding a Pod, and the VolumeManager binding a claim. For example, where does claim binding fit into the Pod lifecycle -- will Kubelet start a Pod that has been assigned a host by the scheduler but hasn't had its claims bound yet? Can VolumeManager bind a claim before scheduler has picked a host? Ideally Pod binding and volume binding would be atomic so that we don't waste resources in the case where a node is available but not a volume, or a volume is available but not a node. To the extent that we're not going to make them atomic, how can we minimize this problem?

Sorry if this was already discussed in the PR thread, which I stopped reading many messages ago.